newsroom/news_api/news/syndicate/__init__.py

```diff
@@ -0,0 +1,72 @@
+import superdesk
+import logging
+import json
+from flask import request, make_response, jsonify
+from eve.methods.get import get_internal
+from .error_handlers import process_error_response
+from .auth import authenticate
+from .syndicate_handlers import FORMAT_HANDLERS, FEED_GENERATORS as FORMAT_HANDLERS_INIT
+from .resource import NewsAPISyndicateResource
+from .service import NewsAPISyndicateService
+from werkzeug.routing import BaseConverter
+
+blueprint = superdesk.Blueprint('syndicate', __name__)
+
+logger = logging.getLogger(__name__)
+
+
+class RegExConverter(BaseConverter):
+    def __init__(self, map, regex='[^/]+'):
+        super().__init__(map)
+        self.regex = regex
+
+
+def get_feed(syndicate_formatter, token=None):
+
+    def generate_feed(syndicate_formatter):
+        response = get_internal('news/search')
+        return FORMAT_HANDLERS_INIT[syndicate_formatter.lower()](response[0])
+
+    return generate_feed(syndicate_formatter)
+
+
+@blueprint.route('/<regex("atom|rss"):syndicate_type>', methods=['GET'])
+@blueprint.route('/<regex("atom|rss"):syndicate_type>/<path:token>', methods=['GET'])
+@authenticate
+def get_syndicate_feed(syndicate_type, token=None):
+    return get_feed(syndicate_type, token)
+
+
+def init_app(app):
+
+    superdesk.register_resource('news/syndicate', NewsAPISyndicateResource, NewsAPISyndicateService, _app=app)
+    app.url_map.converters['regex'] = RegExConverter
+    superdesk.blueprint(blueprint, app)
+
+    @app.after_request
```

Comment:
This function should no longer be required. Formatting the response is already handled in the blueprint view, request validation can be placed in the view, and handling of errors can also be placed there.
This removes the need to have an app global lifecycle hook for one specific view which you already control above.